### PR TITLE
Allow setting of `useElasticSearchOnAws` option on content fetches

### PR DESCRIPTION
**Pull request by ironsidevsquincy**
cc: @matthew-andrews, @charypar

### CODE REVIEW

File: server/lib/graphql.js [hunk 1]
```diff
@@ -24,14 +24,7 @@ const fetch = (backend, opts = {}) => {
...
-
-	const fetchMock = fetch(backend(true, true), opts);
-
-	return {
-		fetch: (mock ? fetchMock : (elastic ? fetchEs : fetchCapi)),
-		printSchema: () => printSchema(schema)
-	}
-};
+export default (opts = {}, flags = {}) => ({
+	fetch: fetch(backend(opts), flags),
```
Comment (charypar) [inline server/lib/graphql.js]: This is going to create a new instance of the query function every time you call it. I suppose it's ok, just making sure we know. That's why it prebuilt them in the previous implementation.

File: server/routes/query.js [hunk 2]
```diff
@@ -14,7 +12,11 @@ export default (req, res, next) => {
 		return res.status(400).send();
 	}
 
-	graphql(useElasticSearch, useMockBackend, { flags })
+	graphql({
+			elasticSearch: flags.elasticSearchItemGet,
+			elasticSearchAws: flags.elasticSearchOnAws,
+			mock: flags.mockFrontPage
+		})
```
Comment (charypar) [inline server/routes/query.js]: Should we pass in `flags` here too?

**Comment by matthew-andrews** [comment]
This flag is only going to be temporary — within a week or two all requests should be going ot the new cluster…… all we need to do now is verify the new cluster is working well enough to your requirements

**Comment by matthew-andrews** [comment]
so maybe we could roll back the new instance stuff then?

**Comment by ironsidevsquincy** [comment]
that's fine, when the flag goes we can just stop passing it through. this change makes it generally easier to pass flags through to the api client so it's worth doing now

**Comment by matthew-andrews** [comment]
:+1:

**Pull request merged**